pkg/utils/utils.go

```diff
@@ -71,3 +82,60 @@ func GetOperatorRuntime() common.Runtime {
 	}
 	return runtimeReturn
 }
+
+// NewPodInformer creates a Shared Index Pod Informer matching the labelSelector string
+func NewPodInformer(kubeClientSet kubernetes.Interface, labelSelectorString string) cache.SharedIndexInformer {
+	return cache.NewSharedIndexInformer(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				return kubeClientSet.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
+					LabelSelector: labelSelectorString,
+				})
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return kubeClientSet.CoreV1().Pods("").Watch(context.Background(), metav1.ListOptions{
+					LabelSelector: labelSelectorString,
+				})
+			},
+		},
+		&corev1.Pod{},  // Resource type
+		time.Second*30, // Resync period
+		cache.Indexers{
+			cache.NamespaceIndex: cache.MetaNamespaceIndexFunc, // Index by namespace
+		},
+	)
+}
+
+// LabelSelectorToString gets a string from a labelSelector
+func LabelSelectorToString(labelSelector metav1.LabelSelector) (string, error) {
+	var matchExpressions []string
+	for _, expr := range labelSelector.MatchExpressions {
+		// Handle only Exists expressions
+		matchExpressions = append(matchExpressions, expr.Key)
+	}
+	// Join match labels and match expressions into a single string with a comma separator.
+	labelSelectorString := strings.Join(matchExpressions, ",")
+	// Validate labelSelectorString
+	if _, err := labels.Parse(labelSelectorString); err != nil {
+		return "", err
+	}
+	return labelSelectorString, nil
+}
```

Comment:
This method is only used from `controller.NewController()`. It's functionality is limited to only supported match expressions (label selectors also support matching on labels). Wouldn't it make sense to remove this function and implement the required code in `controller.NewController()`?

This method implies it can convert any label selector to a string, which is can't.

Reply:
agreed, initially it was meant to support many operations but then it came to just be a single one and agree it doesn't make sense. I removed it and now it's just a string since "key" means "it exists" so it's just a simple string. If we want to support more expressions in the future we can add them.